test/ipld-dag-pb.js

```diff
@@ -95,190 +96,104 @@ module.exports = (repo) => {
-          done()
-        })
-      })
+        expect(node2.remainderPath).to.eql('Data')
+        expect(node2.value).to.eql(cid1.toV0())
 
-      it('resolver.get with option localResolve: true', (done) => {
-        resolver.get(cid3, 'Links/1/Hash/Links/0/Hash/Data', { localResolve: true }, (err, result) => {
-          expect(err).to.not.exist()
-          expect(result.remainderPath).to.equal('Links/0/Hash/Data')
-          expect(result.value).to.eql({
-            '/': 'QmS149H7EbyMuZ2wtEF1sAd7gPwjj4rKAorweAjKMkxr8D'
-          })
-          expect(result.cid).to.deep.equal(cid2)
-          done()
-        })
+        expect(node3.remainderPath).to.eql('')
+        expect(node3.value).to.eql(Buffer.from('I am 1'))
       })
 
-      it('resolver.get value within nested scope (1 level) returns cid of node traversed to', (done) => {
-        resolver.get(cid2, 'Links/0/Hash/Data', (err, result) => {
-          expect(err).to.not.exist()
-          expect(result.cid).to.deep.equal(cid1)
-          done()
-        })
+      it('resolver.get round-trip', async () => {
+        const cid = await resolver.put(node1, multicodec.DAG_PB)
+        const node = await resolver.get(cid)
+        // `size` is lazy, without a call to it a deep equal check would fail
+        const _ = node.size // eslint-disable-line no-unused-vars
+        expect(node).to.deep.equal(node1)
       })
 
-      it('resolver.remove', (done) => {
-        resolver.put(node1, { cid: cid1 }, (err) => {
-          expect(err).to.not.exist()
-          resolver.get(cid1, (err, node) => {
-            expect(err).to.not.exist()
-            remove()
+      it('resolver.remove', async () => {
+        // TODO vmx 2018-12-12: The same repo is used for all tests, there
+        // seems to be some race condition with inserting and removing items.
+        // Hence create a unique item for this test. Though the tests
+        // should really be independent so that there are no race conditions.
+        const createNode = new Promise((resolve, reject) => {
+          const data = Buffer.from('a dag-pb node')
+          dagPB.DAGNode.create(data, (err, node) => {
+            if (err) {
+              return reject(err)
+            }
+            return resolve(node)
           })
         })
-
-        function remove () {
-          resolver.remove(cid1, (err) => {
-            expect(err).to.not.exist()
-            resolver.get(cid1, (err) => {
-              expect(err).to.exist()
-              done()
-            })
-          })
+        const node = await createNode
+        const cid = await resolver.put(node, multicodec.DAG_PB)
+        const sameAsNode = await resolver.get(cid)
+        // `size` is lazy, without a call to it a deep equal check would fail
+        const _ = sameAsNode.size // eslint-disable-line no-unused-vars
+        expect(sameAsNode.data).to.deep.equal(node.data)
+        return remove()
+
+        async function remove () {
```

Comment:
why are these wrapped up in their own function? the two lines should probably just be at the top level of this test

Reply:
The tests were just adapted with minimal changes. The tests are in a bad shape currently and need a bigger refactor (there's lots of duplicated code). But I'd like to do it after this change is merged.